arch/arm/src/stm32/stm32f20xxf40xx_flash.c

```diff
@@ -332,7 +367,7 @@ ssize_t up_progmem_ispageerased(size_t page)
 
 ssize_t up_progmem_eraseblock(size_t block)
 {
-  if (block >= STM32_FLASH_NPAGES)
+  if (block >= PROGMEM_NBLOCKS)
```

Comment:
Referring to https://github.com/messBUS/incubator-nuttx/blob/export/include/nuttx/progmem.h#L142-L162
```
 * Returned Value:
 *   block size or negative value on error.  The following errors are reported
``

This function should return the number of byte (sector size) of the erased sector.

This PR violated that API - and will cause breaking changes to all  calles.